src/cli/action.zig

```diff
@@ -68,17 +76,33 @@ pub const Action = enum {
     pub const help_error = error.ActionHelpRequested;
 
     /// Detect the action from CLI args.
-    pub fn detectCLI(alloc: Allocator) !?Action {
+    /// Pass in a pointer to a LoggingOptions structure if you want to retrieve
+    /// any `-q/--quiet` or `-v/--verbose` options or `null` if you don't need them.
+    pub fn detectCLI(alloc: Allocator, loggingOptions: ?*LoggingOptions) !?Action {
         var iter = try std.process.argsWithAllocator(alloc);
         defer iter.deinit();
-        return try detectIter(&iter);
+        return try detectIter(&iter, loggingOptions);
     }
 
     /// Detect the action from any iterator, used primarily for tests.
-    pub fn detectIter(iter: anytype) Error!?Action {
+    /// Pass in a pointer to a LoggingOptions structure if you want to retrieve
+    /// any `-q/--quiet` or `-v/--verbose` options or `null` if you don't need them.
+    pub fn detectIter(iter: anytype, loggingOptions: ?*LoggingOptions) Error!?Action {
```

Comment:
Is there any reason `loggingOptions` would ever be null outside of test cases?

Reply:
Nope.  I just didn't want to require the tests to pass in a structure when they didn't need to.  I suppose I could create a static one just for testing and have each test use that.

Reply:
I've removed the pointer argument.